src/charm.py

```diff
@@ -45,7 +45,7 @@ def __init__(self, *args):
         self._grafana_agent = COSAgentProvider(
             self,
             metrics_endpoints=[
-                {"path": "/metrics", "port": METRICS_PORT},
+                {"path": "/metrics", "port": str(METRICS_PORT)},
```

Comment:
From looking at the library code I don't think you need to cast this particular variable as a string: https://github.com/canonical/pollen-operator/blob/main/lib/charms/grafana_agent/v0/cos_agent.py#L93-L97